applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp

```diff
@@ -0,0 +1,531 @@
+        N_k(0,2) = 2*normal_patch_node[0]*normal_patch_node[1];
+        T_k1(0,0) = normal_patch_node[0]*normal_patch_node[1];
+        T_k1(0,1) = -normal_patch_node[0]*normal_patch_node[1];
+        T_k1(0,2) = normal_patch_node[1]*normal_patch_node[1]-normal_patch_node[0]*normal_patch_node[0];
+
+        noalias(A1) = prod(trans(p_k),trans(N_k));
+        noalias(A2) = prod(N_k,p_k);
+        noalias(A) += mPenaltyNormal*prod(A1, A2);
+
+        noalias(A1) = prod(trans(p_k),trans(T_k1));
+        noalias(A2) = prod(T_k1,p_k);
+
+        // Finally we add the contributiosn to our local system (A, b)
+        noalias(A) += mPenaltyTangent*prod(A1, A2);
+        // NOTE: Code commented. Could be of interest in the future
+//         noalias(A) += mPenaltyNormal*prod(prod(trans(p_k),trans(N_k)),prod(N_k,p_k));
+//         noalias(A) += mPenaltyTangent*prod(prod(prod(trans(p_k),trans(T_k1)),T_k1),p_k);
+        noalias(b) -= mPenaltyNormal*prod(trans(p_k),trans(N_k))*itPatchNode->GetValue(CONTACT_PRESSURE);
+    }
+
+    // Neighboring nodes:
+    for( auto& i_neighbour_node : itPatchNode->GetValue(NEIGHBOUR_NODES)) {
+        if (i_neighbour_node.Has(CONTACT_PRESSURE)){
+            const array_1d<double, 3>& normal_neigh_node = i_neighbour_node.GetValue(NORMAL);
+            const double x_patch = itPatchNode->X();
+            const double y_patch = itPatchNode->Y();
+            const double x_neigh = i_neighbour_node.X();
+            const double y_neigh = i_neighbour_node.Y();
+            p_k(0,1)= x_neigh-x_patch;
+            p_k(0,2)= y_neigh-y_patch;
+            p_k(1,4)= x_neigh-x_patch;
+            p_k(1,5)= y_neigh-y_patch;
+            p_k(2,7)= x_neigh-x_patch;
+            p_k(2,8)= y_neigh-y_patch;
+            N_k(0,0) = normal_neigh_node[0]*normal_neigh_node[0];
+            N_k(0,1) = normal_neigh_node[1]*normal_neigh_node[1];
+            N_k(0,2) = 2*normal_neigh_node[0]*normal_neigh_node[1];
+            T_k1(0,0) = normal_neigh_node[0]*normal_neigh_node[1];
+            T_k1(0,1) = -normal_neigh_node[0]*normal_neigh_node[1];
+            T_k1(0,2) = normal_neigh_node[1]*normal_neigh_node[1]-normal_neigh_node[0]*normal_neigh_node[0];
+
+            noalias(A1) = prod(trans(p_k),trans(N_k));
+            noalias(A2) = prod(N_k,p_k);
+            noalias(A) += mPenaltyNormal*prod(A1, A2);
+
+            noalias(A1) = prod(trans(p_k),trans(T_k1));
+            noalias(A2) = prod(T_k1,p_k);
+
+            // Finally we add the contributiosn to our local system (A, b)
+            noalias(A) += mPenaltyTangent*prod(A1, A2);
+            noalias(b) += mPenaltyNormal*prod(trans(p_k),trans(N_k))*i_neighbour_node.GetValue(CONTACT_PRESSURE);
+        }
+    }
+
+    // Computing coefficients a: A*a=b
+    KRATOS_INFO_IF("SPRErrorProcess", mEchoLevel > 3) << A << std::endl;
+    
+    Vector coeff(SigmaSize * (TDim+1));
+    Vector b_vector = row(b, 0);
+    Matrix invA;
+    double det;
```

Comment:
i think there is a SolveFunction doing this, which will be more robust in limit cases (i am speaking of mathutils since it is a dense matrix)

Reply:
No, there is not

Reply:
can u add it? using lu_factorize etc, as done in the InvertMatrix but with a vector as input

Reply:
I added a generalized inversion using lu_factorize long time ago, is exactly the same you are proposing. Except you can pass directly the system and it acts as a Solve, but essentially is the same

Reply:
no, doing the inverse first and then multiplying by the inverse is a lot less well conditioned than using directly the lu_factorization in the solution of the system.

what i am telling is to have a function that DOES NOT compute explicitly the inverse, but only the factorization... take a look at

http://boost.2283326.n4.nabble.com/Solving-linear-equations-of-the-form-Ax-b-td2708902.html